### ISSUE TITLE
GitPod prebuilds and ignore submodules

### ISSUE THREAD

**Issue by learner-long-life** [TheEvergreenStateCollege/upper-division-cs#1394]
We can suppress submodule changes from `git status` of the parent module by

```
git config diff.ignoreSubmodules all
```

This needs to be run on every new workspace start, perhaps using GitPod pre-builds.

**Comment by learner-long-life** [comment]
@crowllx can you add this Python package / install command to the GitPod Dockerfile

```
pip3 install html2text
```

With this added to our path

```
export PATH=${PATH}:/home/gitpod/.local/bin
```

We also need to build and install the `pdftotext` utility from `xpdf` source code.

https://www.xpdfreader.com/download.html

Thanks.

**Comment by crowllx** [comment]
Yeah I can do that. Is installing `poppler-utils` not enough for getting pdftotext? what other uses does xpdf have that we need? I was able to get access to pdftotext in gitpod by install poppler via apt.

**Comment by learner-long-life** [comment]
poppler-utils is fine 


Sent from Yahoo Mail for iPhone


On Wednesday, May 8, 2024, 1:39 PM, crowllx ***@***.***> wrote:




Yeah I can do that. Is installing poppler-utils not enough for getting pdftotext? what other uses does xpdf have that we need? I was able to get access to pdftotext in gitpod by install poppler via apt.

**Comment by crowllx** [comment]
I think we want to modify the gitpod.yml file in order to set git config settings as part of the workspace build.
[https://www.gitpod.io/docs/references/gitpod-yml](https://www.gitpod.io/docs/references/gitpod-yml).
I'll try it out later.
